netket/utils/group/_group.py

```diff
@@ -263,50 +264,79 @@ def character_table_readable(self) -> Tuple[List[str], Array]:
         return representatives, self.character_table_by_class
 
     @struct.property_cached
-    def _irrep_matrices(self) -> PyTree:
-        random = np.random.standard_normal
-        # Generate the eigensystem of a random linear combination of matrices
-        # in the group's regular representation (namely, product_table == g)
-        # The regular representation contains d copies of each d-dimensional irrep
-        # all of which return the same eigenvalues with eigenvectors that
-        # correspond to one another in the bases of the several irrep copies
-        e, vs = np.linalg.eig(random(len(self))[self.product_table])
-        e = np.stack((e.real, e.imag))
-        # we only need one eigenvector per eigenvalue
-        _, idx = np.unique(comparable(e), return_index=True, axis=1)
-        vs = vs[:, idx]
-        # We will nedd the true product table as well
+    def _irrep_matrices(self, key=jax.random.PRNGKey(0)) -> List[Array]:
```

Comment:
I need to think how to properly do this.
But I fear this default prngkey here is a sure recipe for hidden bugs...

You are basically saying that unless passed a key, this will always use the same random seed.
Which might make sense, maybe, but I'm unsure...

Reply:
Moreover, this is a property. So you cannot pass the optional kwarg, and it will _always_ have the default value.

Can you try to explain me why exactly you need random numbers here, and what are you trying to achieve?

Reply:
can't we fix the random vector? 
Is there a reason why that _random_ vector must be completely random?

Reply:
The algorithm works by splitting the regular representation (which is known to contain every irrep) into constituent irreps and picking out one for every irrep listed in the character table. This it does by constructing a Hermitian matrix (`e` after the second line of `invariant_subspaces`) that commutes with every matrix of the regular rep. This means that the subspaces spanned by degenerate eigenvectors transform as some rep of the group. We need to make sure that these are all irreps, i.e., there are no accidental degeneracies. The simplest way of doing that is using a random `e` matrix (as long as it commutes with the regular rep matrices). 

The JAX random number generator was included at the request of @femtobit (IDK where that conversation went) to make sure you always get the same irrep matrices, rather than something related by a unitary transformation. IMO we should fix that key once and for all, rather than trying to pass it. (We could also cache a bunch of random numbers from random.org, but this is more transparent.)

Reply:
BTW there's a similar pattern in `character_table_by_class` but there it matters less because the output is deterministic. For irrep matrices, you have the freedom of conjugating all matrices in an irrep with some unitary matrix; randomising the matrix randomises this unitary transformation.

Reply:
> They need to look random to make sure we don't have accidental degeneracies.

But in principle, it should be sufficient to pick a specific random vector, right? I don't mean putting it explicitly in the code, just not exposing an option to change the key and hard-code a specific one (or use numpy with a specific seed, but the JAX API is a bit more comfortable here).

Reply:
Exactly, I'm not sure why I added the key argument, it only makes things less reproducible (and doesn't play well with caching as @PhilipVinc pointed out)